spyder/plugins/variableexplorer/plotlib.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""
+Variable Explorer Plugin plotting library management.
+"""
+
+# Standard library imports
+import importlib
+
+# Local imports
+from spyder.config.base import _
+
+
+# Defining compatible plotting libraries
+# (default library is the first available, see `get_default_plotlib`)
+LIBRARIES = ("matplotlib", "guiqwt")
+
+
+def is_package_installed(modname):
+    """Check if package is installed **without importing it**
+    
+    Note: As Spyder won't start if matplotlib has been imported too early,
+    we do not use `utils.programs.is_module_installed` here because
+    it imports module to check if it's installed.
+    """
+    return importlib.util.find_spec(modname) is not None
+
+
+def get_available_plotlibs():
+    """Return list of available plotting libraries"""
+    return [name for name in LIBRARIES if is_package_installed(name)]
+
+
+def get_default_plotlib():
+    """Return default plotting library"""
+    names = get_available_plotlibs()
+    if names is not None:
+        return names[0]
```

Comment:
This function returns `None` if no library is available, which I think wouldn't look good in our Preferences. So, perhaps it should return `Matplotlib`?

Reply:
Actually that's not a problem because if there is no library available, the combo box won't be displayed in Preferences: combo box is added to Preferences page only if `plotlib.AVAILABLE_PLOTLIBS` is not empty. But I realize that it's not a good idea because even if `plotlib.AVAILABLE_PLOTLIBS` is empty, that does not mean that the IPython kernel won't have the library installed. So I'm rolling back on this.

(in fact, I was anticipating my next move which was to plot `guiqwt` within Spyder process - in the mean time, I did some test embedding Qt widget from another process but it was not quite satisfying in terms of stability)

Reply:
See [7adeece](https://github.com/spyder-ide/spyder/pull/20075/commits/7adeeceea33c88919440c4f2852c5fe1f72811d5)